File: spring-data-jpa/src/main/java/io/beaniejoy/springdatajpa/entity/cafe/Cafe.java
package io.beaniejoy.springdatajpa.entity.cafe;

import io.beaniejoy.springdatajpa.entity.BaseEntity;
import io.beaniejoy.springdatajpa.entity.review.Review;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "cafe")
public class Cafe extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String address;

    private String phoneNumber;

    private String description;

    @Builder.Default
    @OneToMany(mappedBy = "cafe", fetch = FetchType.LAZY)
    private List<Review> reviews = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "cafe", fetch = FetchType.LAZY)
    private List<CafeMenu> cafeMenus = new ArrayList<>();

    public void updateInfo(String name, String address) {
        this.name = name;
        this.address = address;
    }
}

File: spring-boot-v3-batch-demo/src/main/kotlin/io/beaniejoy/batch/domain/entity/Cafe.kt
package io.beaniejoy.batch.domain.entity

import jakarta.persistence.*

@Entity
@Table(name = "cafes")
class Cafe {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    @Column(name = "name")
    val name: String? = null

    @Column(name = "address")
    val address: String? = null

    @Column(name = "phone_number")
    val phoneNumber: String? = null

    @Column(name = "description")
    val description: String? = null

    @OneToMany(mappedBy = "cafe", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
    val cafeMenus: MutableList<CafeMenu> = arrayListOf()
}